src/routes/+page.svelte

```diff
@@ -310,99 +309,149 @@
     handleGroupBy()
   }
 
-  const handleGroupBy = () => {
-    reportGroup = {}
-    auxReportGroup = {}
-
-    if (selectedGroupBy.some((item: FilterOptions) => item.label === 'Date')) {
-      for (const [idIssue, entry] of Object.entries(reportFiltered)) {
-        const dates = new Set(entry.timeEntry.map((item) => formatDayMonthYear(item.end || new Date().toDateString())))
-
-        dates.forEach((date: string) => {
-          if (!reportGroup[date]) {
-            reportGroup[date] = {}
+  /**
+   * Se você vá mexer na função groupHandler e não tá entendendo como funciona, pense como se fosse (não é igual, mas
+   * é pra tentar facilitar o entendimento) um convensor por níveis de uma lista para uma BTree (nesse caso dessa função
+   * irá sempre ter a mesma quantidade de níveis, que é a quantidade de orderBy's), onde:
+   *
+   * @param load é um Nó caso seja do tipo Group ou uma folha se for do tipo Report
+   * @param loadDivisor é a lógica de divisão para cada nivel da BTree
+   * @param isToGroup valida se é necessário fazer o agrupamento desse nível
+   * @param allFieldName o nome do nível caso não precise agrupar
+   * @param loadRoute é o caminho para chegar no Nó/Folha, já que a divisão não segue a mesma lógica em cada nível e
+   * só usar se precisar verificar algo antes desse Nó/Folha
+   *
+   * O retorno é:
+   *  - um nó, caso o retorno seja das folhas
+   *  - parte da árvore, se for uma das funções da recursão
+   *  - a árvore completa, se for a primeira chamada da função
+   */
+  const groupByHandler = (
+    load: Group | Report,
+    loadDivisor: (entry: Entry, route?: string[]) => [string, Entry][],
+    isToGroup: boolean,
+    allFieldName: string,
+    loadRoute: string[] = [],
```

Comment:
Seria melhor se os argumentos estivessem em um único objeto, pq daí fica mais fácil de acompanhar oq está sendo passado de parâmetro, como conversamos recentemente no Discord: https://discord.com/channels/431152129789526016/1107660143010775080/1197237569080021062
```suggestion
    {
      load,
      loadDivisor,
      isToGroup,
      allFieldName,
      loadRoute,
    }: {
      load: Group | Report,
      loadDivisor: (entry: Entry, route?: string[]) => [string, Entry][],
      isToGroup: boolean,
      allFieldName: string,
      loadRoute: string[] = []
    }
```